includes/payments/functions.php

```diff
@@ -579,6 +579,34 @@ function edd_check_for_existing_payment( $payment_id ) {
 }
 
 /**
+ * Retrieves an instance of EDD_Payment for a specified ID.
+ *
+ * @since 2.7
+ *
+ * @param int $id Payment ID.
+ * @return mixed false|object EDD_Payment if a valid payment ID, false otherwise.
+ */
+function edd_get_payment( $payment_id = 0 ) {
```

Comment:
This should also support passing in a `WP_Post` object and an `EDD_Payment` object.